### PR TITLE
add TableOfContents for mdx

### PR DESCRIPTION
**Pull request by infiniteluke**
Needs styling.

Usage:
add

```
<TableOfContents />
```

to your mdx to see it show the table of contents for the mdx file you're editing.

example

```mdx
---
title: "CTAs"
description: "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua."
tab: "Code"
tabOrder: 1
publishToStyleGuide: true
---

<TableOfContents />

# This is a CTA

This is a CTA that uses a button in it.

## Example

<Component />

## Source

<Code />
```

### CODE REVIEW

File: src/components/Templates/layout.js [hunk 1]
```diff
@@ -8,14 +8,48 @@ import "./layout.css";
 // Components for MDX
 import ComponentViewer from "../Atoms/ComponentViewer/ComponentViewer.component";
 import CodeSnippet from "../Atoms/CodeSnippet/CodeSnippet.component";
+import TableOfContents from "../Molecules/TableOfContents/TableOfContents.component";
+
+const HEADINGS = [
```
Comment (infiniteluke) [inline src/components/Templates/layout.js]: This is necessary so that we can add the anchor links

File: src/components/Templates/layout.js [hunk 2]
```diff
@@ -8,14 +8,48 @@ import "./layout.css";
...
+];
+
+const Heading = level => ({ props, children }) => {
+  const H = HEADINGS[level - 1];
+  const name =
+    typeof children === "string"
+      ? children.replace(/\s+/g, "-").toLowerCase()
+      : "";
+  return (
+    <a name={name} href={`#${name}`}>
```
Comment (infiniteluke) [inline src/components/Templates/layout.js]: This is how we add the anchor links for the table of contents

File: src/components/Molecules/TableOfContents/table-of-contents.css [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  margin-bottom: 0.75rem;
+}
+
+.toc__list {
+  margin: 0;
+}
+
+.toc__item {
+  list-style: none;
+  margin: 0;
```
Comment (infiniteluke) [inline src/components/Molecules/TableOfContents/table-of-contents.css]: This isn't indenting quite right

**Comment by amazingrando** [comment]
Styled the TOC. I've also made all the border-radii consistent.

Screenshot 2019-07-29 16 31 22

**Pull request merged**